frontend/src/hooks/useTrafficData.ts

```diff
@@ -132,7 +132,7 @@ const toChartData = (
 
             for (const dayKey in item.days) {
                 const day = item.days[dayKey];
-                const dayNum = new Date(Date.parse(day.day)).getDate();
+                const dayNum = new Date(Date.parse(day.day)).getUTCDate();
```

Comment:
I'd add a test for the function generating those dates that fails with previous code and succeeds with the new one

Reply:
He showed me the test but it needs to run vite with a different timezone

Reply:
Thanks, yep fully onboard with that @kwasniew - but I'm yet to find a good way to have a committable repro test for this :( Looking into spoofing timezones and such, but doesn't bite.

Reply:
may not be worth the effort. ideally before the test we should change the timezone and reset to original value afterward.

Reply:
another option would be to avoid overriding the timezone in the tested method explicitly

Reply:
```ts
beforeEach(() => {
  // Set the timezone to the desired value
  process.env.TZ = 'America/New_York';
});

afterEach(() => {
  // Reset the timezone to the original value
  process.env.TZ = originalTimeZone;
});
```

Reply:
Trying that but can't get it to "bite" unfortunately